### ISSUE TITLE
Configuring Multiple Tenants

### ISSUE THREAD

**Issue by Geraint-Davies** [getfider/fider#1128]
I don't know if this is released/stable functionality that just hasn't been documented, of if it's still in beta, but does anyone have a working example of a fider instance with more than one tenant?

I've seen a comment in [this issue](https://github.com/getfider/fider/issues/822) that mentions adding two new environment variables, but that doesn't seem to work for me.

I have added: `HOST_MODE: multi` and `HOST_DOMAIN` with my hostname (tried with and without the leading `https://` but left it in **without** the prefix).

The result is expected to be:

> That'll turn your app into multi-tenant. Each tenant will have its url as <tenant subdomain>.yoursite.com . You can then later add a custom CNAME if you wish.

but after updating the subdomain value of the tenants table to have a value other than default it doesn't work.

The result I get is the Fider `Page not found` page now when I visit the hostname. And an unhandled `Hmmm... can't reach this page. The server IP address could not be found` when I try my subdomain.hostname.net.

I've self hosted, in Azure, using the latest stable docker image getfider/fider:stable

Thanks in advance!

**Comment by reubence** [comment]
Did you get it working @Geraint-Davies?

**Comment by Erwane** [comment]
In docker compose App environment:
```yml
    environment:
      BASE_URL: https://feedback.net/
      HOST_MODE: multi
      HOST_DOMAIN: feedback.net
```
(restart or rebuild)

In database

```sql
update tenants set subdomain='sub' where id=1;
```

Tenant is https://sub.feedback.net/

This is working for one sub tenant but i can't find how to create tenants.
The api endpoint is ` POST _api/tenants` but no documentation.

By injecting tenants in database, it's working.

 ```sql
INSERT INTO tenants (name,  subdomain, created_at, cname, invitation, welcome_mesage, status, is_private, custom_css, logo_bkey, locale, is_email_auth_allowed, is_feed_enabled, prevent_indexing)
VALUES ('Sub2', 'sub2',    now(),      '',     'Your suggestion ...',  'Welcome Message', 1, false, '', '', 'en', true, false, true)
```
Should duplicates your admin user to

```sql
INSERT INTO users (name, email, created_at, tenant_id, role, status, api_key, api_key_date, avatar_type, avatar_bkey, email_supressed_at) VALUES ('Me', 'me@example.com', now(), 2, 3, 1, 'RegenerateApiKeyAfterLogin', now(), 1, '', null)
```
now login to https://sub2.feedback.net/ "et voilà"

**Comment by mattwoberts** [comment]
> In docker compose App environment:
> 
>     environment:
>       BASE_URL: https://feedback.net/
>       HOST_MODE: multi
>       HOST_DOMAIN: feedback.net
> 

You're off to a good start. You need to set these 3 env vars to tell Fider to run in "multi" mode. You don't need to do anything with your database. Just set these 3 vars.

Once you've done that, if you want a new instance, you go to `https://login.feedback.net/signup` - if all is working well that should present you with a signup page for your new instance.

**Comment by Erwane** [comment]
Wow, very nice.
Where is the documentation for that please ?

and ~how to manages tenants or allow signup on one tenant but not to create new tenant ?~ vhost configuration do that

**Comment by mattwoberts** [comment]
> Wow, very nice. Where is the documentation for that please ?
> 
> and ~how to manages tenants or allow signup on one tenant but not to create new tenant ?~ vhost configuration do that

Yeah the docs are pretty much non-existent for this - it's a todo that's a bit lacking, PRs welcome :)

**Comment by Geraint-Davies** [comment]
Thanks for revisiting this!! 😊🥳
What if your site is Azure hosted so the domain is already `https://my-fider-demo.azurewebsites.net` ? 
Then should that work as:  `https://sub.my-fider-demo.azurewebsites.net` ? Or does the base have to be a root domain address?

What I'm experiencing is after I update the HOST_MODE and HOST_DOMAIN, then the base URL now turns to "PAGE NOT FOUND" and the subs don't work either and says "Hmmm… can't reach this page".

**Comment by Geraint-Davies** [comment]
FYI @mattwoberts `https://login.feedback.net/signup` doesn't work for me even when my hosted instance _isn't_ in multi mode. (where instead of feedback.net, it's my Azure URL, similar to `my-fider-demo.azurewebsites.net`). Is this a bug? Or something that needs configuring on an Azure webservice to allow sub-domains?

**Comment by mattwoberts** [comment]
@Geraint-Davies I'm not sure if it's something that's related to your azure config to be honest, however the login.feedback.net thing is only setup when you're in multi mode, I wouldn't expect that to work when you're not not in multi. Hope you get this working...

**Comment by Geraint87** [comment]
@mattwoberts - thanks, I'll keep trying. Any idea what config fider might depend on for this? I haven't customised the resource - just deployed it using the steps in [the docs](https://docs.fider.io/hosting-on-azure).

**Comment by StefanOltmann** [comment]
> Once you've done that, if you want a new instance, you go to https://login.feedback.net/signup

That raises the question of how I can prevent people who know I run Fider from creating new tenants.

**Comment by shivenigma** [comment]
@mattwoberts  I submitted a PR for the multi-tenant documentation and happy to improve/change if you think the docs needs refinement. Would be great if you could take time to take a look.

**Comment by shivenigma** [comment]
@mattwoberts I know you could be overwhelmed, but is there someone else that can review the documentation update PR?
